pkg/pillar/containerd/containerd.go

```diff
@@ -41,10 +42,10 @@ import (
 )
 
 const (
-	// EVE persist storage type file (content can be: ext3, ext4, zfs)
-	eveStorageTypeFile = "/run/eve.persist_type"
 	// containerd socket
 	ctrdSocket = "/run/containerd/containerd.sock"
+	// containerd user socket
+	ctrdUserSocket = "/run/containerd-user/containerd.sock"
```

Comment:
If I understood correctly, then inside pillar, there is only one containerd ever to access, the user one. The system one is the one that runs pillar. Pillar (I think) would never have reason to access the system one. So it won't be mounted in or accessible. Could we not then make this just `ctrdSocket = "/run/containerd/containerd.sock"` (which is the user one, but as far as pillar is concerned, the only one)? 

That also would change the `waitForUserContainerd()` to just `waitForContainerd()`. It also would simplify many of these changes, as they still would just look the same: start/stop/remove/etc containers, and load/unload images, all in the single containerd that pillar knows about.

Is there a reason that pillar still needs access to the system one?

Reply:
Right now user containerd is here to be used by CAS (download/prepare data in /persist/vault/volumes). System containerd still here to run domains

Reply:
Is there any reason to continue to do so? From the pillar code's perspective, containerd is containerd. It shouldn't care. If you already are segregating (and I still think this is a good approach you have), then might as well do it completely.

Reply:
We want to communicate with xen-tools running in system containerd in hypervisor. And in general run domains based on xen-tools we build inside linuxkit (system containerd).

Reply:
Isn't hypervisor control inside pillar as well? You started segregation, might as well go all the way, unless there is a reason we cannot.

Reply:
The problem I can see: hypervisor module for xen wants to talk with "xen-tools" container which is inside "system" containerd in many places ([for example](https://github.com/lf-edge/eve/blob/e2c144539ea2dfbb820e19f8c612b84e9166a768/pkg/pillar/hypervisor/xen.go#L569) when we work with AssignableAdapters) and as @eriknordmark mentioned https://github.com/lf-edge/eve/pull/2524#discussion_r811426341 we want to manage adapters before/in parallel with vault unlock, i.e. before start of the second containerd.

Reply:
> hypervisor module for xen wants to talk with "xen-tools" container which is inside "system" containerd in many places

I am not sure I understand this. Looking at the example to which you linked, I think you are saying that:

1. `xen-tools` runs as a standalone system-level container, parallel to pillar (see [here](https://github.com/lf-edge/eve/blob/1c8a58d227ab955639273552771ff5f1eae2ab5f/images/rootfs.yml.in#L61-L64))
2. `hypervisor` module (which is part of `pillar`) needs to do things to various VM instances
3. `hypervisor` does so by exec-ing into the running system-level container
4. Therefore `hypervisor` needs to be have access to the system-level containerd, so it can do the exec-into tasks

Is that it? If so, is it only hypervisor and its xen-tools requirements? Or are there other areas that also need access to system-level?

> as @eriknordmark mentioned https://github.com/lf-edge/eve/pull/2524#discussion_r811426341 we want to manage adapters before/in parallel with vault unlock, i.e. before start of the second containerd.

I don't understand. What does this have to do with system vs workload containerd?

Reply:
> Is that it? If so, is it only hypervisor and its xen-tools requirements? Or are there other areas that also need access to system-level?

Yes, and for this reason I use system containerd right now.

> I don't understand. What does this have to do with system vs workload containerd?

I just mention expected sequence, we want to interact with system containerd before vault unlocked and we want to run user containerd after vault unlocked.

Reply:
Looking into it, I see that `CtrSystemExec()` is called 7 times in [xen.go](https://github.com/lf-edge/eve/blob/e2c144539ea2dfbb820e19f8c612b84e9166a768/pkg/pillar/hypervisor/xen.go):

* `xl dmesg` - 2x
* `xl destroy` - 1x
* `xl pci-assignable-add` - 1x
* `xl pci-assignable-rem` - 1x
* `xl info` - 1x
* `xentop -b -d 1 -I 2 -f` - 1x

Is that the total of it?

Reply:
> in general we use /containers/services/xen-tools prepared by linuxkit to start our domains

Do you mean other than the above 7 places? I didn't see anything there to start a domain.

I should remember all of this. 🤷‍♂️

Reply:
Oh right, [here](https://github.com/lf-edge/eve/blob/e2c144539ea2dfbb820e19f8c612b84e9166a768/pkg/pillar/hypervisor/xen.go#L131-L155). That it?

Reply:
Having had a chance to think this one through, I still agree with your approach. I also think we should do complete segregation: user containerd inside pillar to start userspace both VM domains and containers, system containerd to start system-level services, which pillar will not even be aware of. FWIW, rancheros was doing this years ago, and docker for Mac/windows does something a bit similar.

That having been said, we should not hold this up for that. We should get this in, then put in the effort to figure out how to get all of the userspace workloads entirely separate from system